### ISSUE TITLE
File glob doesn't include base directory

### ISSUE THREAD

**Issue by mockdeep** [stylelint/stylelint#2243]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This appears to be a bug where `stylelint` doesn't pick up files in the base directory of the glob. For example, if I have a file structure like:

```
css/
  main.scss
  variables/_colors.scss
```

When I try to run the command like the following, it only runs on `_colors.scss`:

```sh
stylelint css/**/*.scss
```

> Which version of stylelint are you using?

e.g. `7.7.1`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS.

> What did you expect to happen?

I expected it to also run on `main.css`

> What actually happened (e.g. what warnings or errors you are getting)?

It only runs on files in sub-directories of the glob expression.

**Comment by mockdeep** [comment]
Actually, it looks like maybe the glob stops looking if it finds files in the current path, so if I go one level higher in this case it'll only pick up `main.scss` and not the further nested stuff:

```sh
stylelint ./**/*.scss
```

However, if I go one level higher, it seems to pick everything up:

```sh
stylelint ../**/*.scss
```

**Comment by jeddy3** [comment]
@mockdeep Thanks for the report.

I can reproduce this. 

Can someone with more experience than I in globs confirm that:

```
stylelint css/**/*.css
```

Should check all three of the css files in this directory structure:

```console
css/
  1.css
  sub-directoy/
    2.css
    sub-sub-directory/
      3.css
```

Currently it looks like only `2.css` is checked.

**Comment by davidtheclark** [comment]
If you are using *unquoted* globs, they are being resolved by your particular shell. What shell are you using @mockdeep and @jeddy3?

In `zsh`, I am *not* able to reproduce this, as evidenced by the output of the following command:

```
➜  stylelint git:(master) node lib/cli.js --config lib/__tests__/fixtures/config-no-pixels.json --formatter verbose lib/__tests__/fixtures/**/*.css 

lib/__tests__/fixtures/needlessDisables/disabled-ranges-3.css
 1:1  ✖  Unclosed block   CssSyntaxError

13 sources checked
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/config-per-file/a/a.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/config-per-file/a/b/b.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/config-per-file/a/b/c/c.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/config-per-file/a/b/c/d/d.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/empty-block-with-disables.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/empty-block.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/getConfigForFile/a/b/foo.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/invalid-hex.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/needlessDisables/disabled-ranges-1.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/needlessDisables/disabled-ranges-2.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/needlessDisables/disabled-ranges-3.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/needlessDisables/ignored-file.css
 /Users/davidclark/dev/stylelint/lib/__tests__/fixtures/standaloneNoParsing/no-syntax-error.css

1 problem found
 severity level "error": 1
  CssSyntaxError: 1
```

`/Users/davidclark/dev/stylelint/lib/__tests__/fixtures/invalid-hex.css` is in fact read.

However, that's just investigation to clarify what you're seeing. The truth is that stylelint has no control over unquoted globs, so you'll need to learn what your shell does. 

*And/or* you can use *quoted* globs, which stylelint *does* control, turning them over to `globby`. I just tried the same command as above with quoted globs, and it also worked.

**Comment by jeddy3** [comment]
@davidtheclark Thanks! Of course. I feel like a bit of a nincompoop for not realising this. It's the reason we [wrapped all the globs in quotes in the examples](https://github.com/stylelint/stylelint/commit/8e66c0c26a30c0c3223e566bc0e07f8c3f656387) in the [user guide](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md).

It's also in the first line of the CLI `--help`

> If an input argument is wrapped in quotation marks, it will be passed to node-glob for cross-platform glob support.

**Comment by mockdeep** [comment]
Ah, okay. I'm using bash.
